paimon-core/src/main/java/org/apache/paimon/manifest/ManifestFileMeta.java

```diff
@@ -185,4 +207,108 @@ private static void mergeCandidates(
             newMetas.addAll(merged);
         }
     }
+
+    static class FullCompactionHelper {
```

Comment:
We don't need a class, just a method is OK.
```
public Optional<Integer> tryPickFullCompaction(...) {
}
```

Returns index to do full compaction.